### PR TITLE
ddl, metrics: remove cleanExpirePath

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

We implemented CleanExpiredPath because of the etcd's bug. Since it had been fixed a long time ago, we can remove it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

**Review by ngaut — APPROVED** [review]

**Comment by ngaut** [comment]
Could you post a link about the the bug reference of etcd?

File: ddl/ddl.go [hunk 1]
```diff
@@ -659,7 +659,6 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
 	d.prepareWorkers4ConcurrencyDDL()
 	d.prepareWorkers4legacyDDL()
 
-	go d.schemaSyncer.StartCleanWork()
```
Comment (zimulala) [inline ddl/ddl.go:662]: We need to remove `metrics.DDLCounter.WithLabelValues(metrics.StartCleanWork).Inc()` in line667.

File: ddl/util/syncer.go [hunk 2]
```diff
@@ -115,19 +108,17 @@ type schemaVersionSyncer struct {
...
 	return &schemaVersionSyncer{
-		etcdCli:                   etcdCli,
-		selfSchemaVerPath:         fmt.Sprintf("%s/%s", DDLAllSchemaVersions, id),
-		ownerChecker:              oc,
-		notifyCleanExpiredPathsCh: make(chan struct{}, 1),
-		ctx:                       childCtx,
-		cancel:                    cancelFunc,
+		etcdCli:           etcdCli,
+		selfSchemaVerPath: fmt.Sprintf("%s/%s", DDLAllSchemaVersions, id),
+		ownerChecker:      oc,
```
Comment (zimulala) [inline ddl/util/syncer.go:119]: Could we remove `ownerChecker` too?

File: metrics/ddl.go [hunk 3]
```diff
@@ -70,13 +70,10 @@ var (
 			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20), // 1ms ~ 524s
 		}, []string{LblResult})
 
-	OwnerUpdateGlobalVersion    = "update_global_version"
-	OwnerGetGlobalVersion       = "get_global_version"
-	OwnerCheckAllVersions       = "check_all_versions"
-	OwnerNotifyCleanExpirePaths = "notify_clean_expire_paths"
-	OwnerCleanExpirePaths       = "clean_expire_paths"
-	OwnerCleanOneExpirePath     = "clean_an_expire_path"
-	OwnerHandleSyncerHistogram  = prometheus.NewHistogramVec(
```
Comment (zimulala) [inline metrics/ddl.go:79]: Please remove `StartCleanWork` in [line105](https://github.com/pingcap/tidb/pull/36863/files#diff-e880516abc91d29e0ea2a8619091e50fd31338e2a9fae2d4da360662b8d5cbe1L105)

**Review by zimulala — CHANGES_REQUESTED** [review]

**Comment by zimulala** [comment]
> Could you post a link about the the bug reference of etcd?

[The key is not deleted when the bound lease expires](https://github.com/etcd-io/etcd/issues/10686)

**Review by zimulala — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**